source/bluetooth/MicroBitPartialFlashingService.cpp

```diff
@@ -289,16 +338,29 @@ void MicroBitPartialFlashingService::partialFlashingEvent(MicroBitEvent e)
        
        KeyValuePair* flashIncomplete = storage.get("flashIncomplete");
        if(flashIncomplete == NULL){
+
          uint8_t flashIncompleteVal = 0x01;
          storage.put("flashIncomplete", &flashIncompleteVal, sizeof(flashIncompleteVal));
+            
+         // Invalidate filesystem
+         MicroBitFileSystem* fs;
+         if(MicroBitFileSystem::defaultFileSystem == NULL) {
+             fs = new MicroBitFileSystem();
+         } else {
+             fs = MicroBitFileSystem::defaultFileSystem;
+         }
+         fs->reformat();
        }
        delete flashIncomplete;
 
       uint32_t *flashPointer   = (uint32_t *)(offset);
 
       // If the pointer is on a page boundary erase the page
-      if(!((uint32_t)flashPointer % MICROBIT_CODEPAGESIZE))
-          flash.erase_page(flashPointer);
+      if(!((uint32_t)flashPointer % MICROBIT_CODEPAGESIZE)) {
+          if(crc32_compute((uint8_t*)flashPointer, MICROBIT_CODEPAGESIZE, NULL) != 0xF154670A) {
```

Comment:
Checking the page for any non-FFs would be more robust.

Reply:
In the latest commit it is checking for non-FF

Reply:
Sorry! I must have got hooked into the wrong version. This looks good now.